File: dotnet/Operon.Sdk/Models/Signature.cs
using System.Text.Json.Serialization;

namespace Operon.Sdk.Models;

/// <summary>
/// Represents a digital signature attached to an Operon transaction payload.
/// </summary>
public sealed class Signature
{
    [JsonPropertyName("algorithm")]
    public string Algorithm { get; set; } = "EdDSA";

    [JsonPropertyName("value")]
    public string? Value { get; set; }
        = string.Empty;

    [JsonPropertyName("keyId")]
    public string? KeyId { get; set; }
        = string.Empty;

    public Signature Clone() => new()
    {
        Algorithm = Algorithm,
        Value = Value,
        KeyId = KeyId
    };
}

File: java/src/main/java/cloud/operon/sdk/Signature.java
package cloud.operon.sdk;

/**
 * Encapsulates a digital signature suitable for Operon Client API submissions. The payload hash is supplied as part
 * of {@link TransactionRequest}; this record captures the algorithm used, the signature value (base64 encoded), and
 * the key identifier. {@code keyId} is optional—when self-signing is enabled the SDK will populate it automatically.
 *
 * @param algorithm signing algorithm identifier (for example {@code EdDSA}).
 * @param value     base64 encoded signature material.
 * @param keyId     optional key identifier referencing the signing key.
 */
public record Signature(String algorithm, String value, String keyId) {

    /**
     * Returns a copy with the key id replaced. Useful for callers who want to reuse the same signature but attach
     * a different key reference without constructing a new {@link Signature} manually.
     */
    public Signature withKeyId(String keyId) {
        return new Signature(algorithm, value, keyId);
    }
}
